### PR TITLE
Fix datetime imports

### PR DESCRIPTION
**Pull request by pnuu**
In many places, we have used `from datetime import datetime`. This is bad, since it masks the module for further imports:

```python
from datetime import datetime
from datetime.timezone import utc
```

Sorting these the other way around will cause complaints from flake8/ruff.

This PR changes all of these import to `import datetime as dt`, so we need on the single import to access other items:

```python
import datetime as dt

dt.datetime(1980, 3, 19)
dt.timedelta(minutes=42)
now = dt.datetime.now(dt.timezone.utc)
```

This is a change that makes implementing the timezone aware datetimes of https://github.com/pytroll/satpy/issues/2752 easier.

 - [ ] Closes #xxxx 
 - [ ] Tests added 
 - [ ] Fully documented 
 - [ ] Add your name to `AUTHORS.md` if not there already

### CODE REVIEW

File: satpy/tests/reader_tests/test_seviri_l1b_icare.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def test_load_dataset_vis(self):
         datasets = r.load(["VIS008"])
         assert len(datasets) == 1
         for v in datasets.values():
-            dt = datetime(2004, 12, 29, 12, 27, 44)
+            dt = dt.datetime(2004, 12, 29, 12, 27, 44)
```
Comment (sfinkens) [inline satpy/tests/reader_tests/test_seviri_l1b_icare.py:136]: This overrides the module with a timestamp

**Review by sfinkens — CHANGES_REQUESTED** [review]
Oh, nasty... Thanks for fixing this!

Variables previously named `dt` are now called `date`. I think this might be misleading in some cases. Sometimes it's a date, sometimes a timestamp with more precision, sometimes an offset or a timedelta. Obviously the previous name wasn't any better, so feel free to ignore this if takes too much time :)

File: satpy/tests/reader_tests/test_seviri_l1b_icare.py (hunk 1 above)
Comment (pnuu) [reply satpy/tests/reader_tests/test_seviri_l1b_icare.py]: That only fixed the one in the snippet, I'll have a look at what each of the `dt` -> `date` actually mean and update accordinly.

Comment (pnuu) [reply satpy/tests/reader_tests/test_seviri_l1b_icare.py]: And the somewhat better naming in https://github.com/pytroll/satpy/pull/2787/commits/37301177ea7bded84500ca596c82e621a1b1ef4d

**Review by sfinkens — APPROVED** [review]
Thanks for all the hard work!

**Pull request merged**